src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3001,6 +3010,35 @@ private BoundExpression BindStackAllocArrayCreationExpression(StackAllocArrayCre
             return new BoundStackAllocArrayCreation(node, count, pointerType, hasErrors || typeHasErrors);
         }
 
+        private bool IsLegalStackAllocLocation(StackAllocArrayCreationExpressionSyntax node)
+        {
+            if (node.Parent == null ||
+                node.Parent.Kind() != SyntaxKind.EqualsValueClause ||
```

Comment:
>node.Parent [](start = 16, length = 11)

Consider adding a local for `parent` and then updating it in your condition as you look further up the tree. `(parent = node.Parent).Kind() != SyntaxKind.xyz ||`